arduino-ide-extension/src/browser/theia/workspace/workspace-input-dialog.ts

```diff
@@ -14,12 +14,17 @@ export class WorkspaceInputDialog extends TheiaWorkspaceInputDialog {
   constructor(
     @inject(WorkspaceInputDialogProps)
     protected override readonly props: WorkspaceInputDialogProps,
-    @inject(LabelProvider) protected override readonly labelProvider: LabelProvider
+    @inject(LabelProvider)
+    protected override readonly labelProvider: LabelProvider
   ) {
     super(props, labelProvider);
-    this.appendCloseButton(
-      nls.localize('vscode/issueMainService/cancel', 'Cancel')
-    );
+    if (this.acceptButton) {
+      this.controlPanel.removeChild(this.acceptButton);
+      this.appendCloseButton(
+        nls.localize('vscode/issueMainService/cancel', 'Cancel')
+      );
+      this.appendAcceptButton();
+    }
```

Comment:
It feels weird to me to add the accept button first, then remove it and then add it again only to add the close button before it.

I would prefer to override `this.appendCloseButton` with something like this:
```
  private override appendCloseButton(text: string): HTMLButtonElement {
    this.closeButton = this.createButton(text);
    this.controlPanel.insertBefore(this.closeButton, this.acceptButton);
    this.closeButton.classList.add('secondary');
    return this.closeButton;
  }
```
This should work correctly if you change back the constructor code to the previous state
```
this.appendCloseButton(
      nls.localize('vscode/issueMainService/cancel', 'Cancel')
    );
```

Reply:
I tried to do this using both `insertBefore` and `prepend` but failed.

The solution you are proposing is not working as it is, I did another change [here](https://github.com/arduino/arduino-ide/pull/1382/commits/c7cdaa6ef0585ec33f1ba22a4e83f4cad8fe7525). Let me know if it could be better.

Reply:
Looks just fine to me, thanks Franceso!